src/cli/index.ts

```diff
@@ -42,16 +43,33 @@ export default class Cli {
   private readonly argv: string[]
   private readonly cwd: string
   private readonly stdout: IFormatterStream
+  private readonly console: Console
 
-  constructor({ argv, cwd, stdout }) {
+  constructor({ argv, cwd, stdout, stderr }) {
     this.argv = argv
     this.cwd = cwd
     this.stdout = stdout
+    this.console = new Console(stderr)
   }
 
   async getConfiguration(): Promise<IConfiguration> {
-    const fullArgv = await getExpandedArgv({ argv: this.argv, cwd: this.cwd })
-    return ConfigurationBuilder.build({ argv: fullArgv, cwd: this.cwd })
+    const fullArgv = await getExpandedArgv({
+      argv: this.argv,
+      cwd: this.cwd,
+    })
+    this.lintArgv(fullArgv)
+    return ConfigurationBuilder.build({
+      argv: fullArgv,
+      cwd: this.cwd,
+    })
+  }
+
+  private lintArgv(fullArgv: string[]): void {
+    if (fullArgv.includes('--retryTagFilter')) {
+      this.console.warn(
+        'the argument --retryTagFilter is deprecated and will be removed in a future release; please use --retry-tag-filter'
+      )
+    }
```

Comment:
Could we put this in the argvParser? 

If we do that, I'm uncertain its worth trying to plumb through stderr to there, but I'm also not opposed to that. Alternatively, we could just use `@spawn` in the feature test

Reply:
I did have it as part of the `parse` method there originally, but the warning would be logged twice if you used the camelCased version from the command line, because `parse` is invoked once with just the argv from the command line, then again with the ones from the resolved profile prepended.

Reply:
We could add an option to the argvParser to disable linting for when we are loading just the profile

Reply:
Made some changes, now in a `lint` method on the parser which is called at the right moment, and using `@spawn` in the scenario so we don't have to create a console and pass it around.